### ISSUE TITLE
unix-dgram does not seem to work with Node v12?

### ISSUE THREAD

**Issue by aizkhaj** [winstonjs/winston-syslog#124]
I tried searching for an issue related to `unix-dgram` and came across this issue, seems related #99. I can only infer node v12 is unsupported here?
Current versions:
Node: 12.8.1
Yarn: 1.19.0

Getting the following logs upon install of this plugin:

```
-> yarn add winston-syslog
yarn add v1.19.0
warning package.json: No license field
warning api@0.0.0: No license field
[1/6] 🔍  Validating package.json...
warning api@0.0.0: No license field
[2/6] 🔍  Resolving packages...
[3/6] 🚚  Fetching packages...
[4/6] 🔗  Linking dependencies...
warning " > feathers-mongoose@8.1.0" has incorrect peer dependency "mongoose@^5.7.3".
warning " > winston-azure-application-insights@2.0.0" has incorrect peer dependency "applicationinsights@~1.0.6".
warning " > eslint-config-airbnb-base@13.2.0" has incorrect peer dependency "eslint@^4.19.1 || ^5.3.0".
[5/6] 🔨  Building fresh packages...
[-/3] ⠁ waiting...
[-/3] ⠁ waiting...
warning Error running install script for optional dependency: "/Users/zaz/D**s/A**s/api/node_modules/unix-dgram: Command failed.
Exit code: 1
Command: node-gyp rebuild
Arguments: 
Directory: /Users/zaz/D**s/A**s/api/node_modules/unix-dgram
Output:
gyp info it worked if it ends with ok
gyp info using node-gyp@5.0.3
gyp info using node@12.8.1 | darwin | x64
gyp info find Python using Python version 2.7.16 found at \"/usr/local/opt/python@2/bin/python2.7\"
gyp info spawn /usr/local/opt/python@2/bin/python2.7
gyp info spawn args [
gyp info spawn args   '/Users/zaz/.nvm/versions/node/v12.8.1/lib/node_modules/npm/node_modules/node-gyp/gyp/gyp_main.py',
gyp info spawn args   'binding.gyp',
gyp info spawn args   '-f',
gyp info spawn args   'make',
gyp info spawn args   '-I',
gyp info spawn args   '/Users/zaz/D**s/A**s/api/node_modules/unix-dgram/build/config.gypi',
gyp info spawn args   '-I',
gyp info spawn args   '/Users/zaz/.nvm/versions/node/v12.8.1/lib/node_modules/npm/node_modules/node-gyp/addon.gypi',
gyp info spawn args   '-I',
gyp info spawn args   '/Users/zaz/Library/Caches/node-gyp/12.8.1/include/node/common.gypi',
gyp info spawn args   '-Dlibrary=shared_library',
gyp info spawn args   '-Dvisibility=default',
gyp info spawn args   '-Dnode_root_dir=/Users/zaz/Library/Caches/node-gyp/12.8.1',
gyp info spawn args   '-Dnode_gyp_dir=/Users/zaz/.nvm/versions/node/v12.8.1/lib/node_modules/npm/node_modules/node-gyp',
gyp info spawn args   '-Dnode_lib_file=/Users/zaz/Library/Caches/node-gyp/12.8.1/<(target_arch)/node.lib',
gyp info spawn args   '-Dmodule_root_dir=/Users/zaz/D**s/A**s/api/node_modules/unix-dgram',
gyp info spawn args   '-Dnode_engine=v8',
gyp info spawn args   '--depth=.',
gyp info spawn args   '--no-parallel',
gyp info spawn args   '--generator-output',
gyp info spawn args   'build',
gyp info spawn args   '-Goutput_dir=.'
gyp info spawn args ]
gyp info spawn make
gyp info spawn args [ 'BUILDTYPE=Release', '-C', 'build' ]
  CXX(target) Release/obj.target/unix_dgram/src/unix_dgram.o
../src/unix_dgram.cc:118:8: warning: 'MakeCallback' is deprecated [-Wdeprecated-declarations]
  Nan::MakeCallback(Nan::GetCurrentContext()->Global(),
       ^
../../nan/nan.h:1024:3: note: 'MakeCallback' has been explicitly marked deprecated here
  NAN_DEPRECATED inline v8::Local<v8::Value> MakeCallback(
  ^
../../nan/nan.h:104:40: note: expanded from macro 'NAN_DEPRECATED'
# define NAN_DEPRECATED __attribute__((deprecated))
                                       ^
../src/unix_dgram.cc:127:8: warning: 'MakeCallback' is deprecated [-Wdeprecated-declarations]
  Nan::MakeCallback(Nan::GetCurrentContext()->Global(),
       ^
../../nan/nan.h:1024:3: note: 'MakeCallback' has been explicitly marked deprecated here
  NAN_DEPRECATED inline v8::Local<v8::Value> MakeCallback(
  ^
../../nan/nan.h:104:40: note: expanded from macro 'NAN_DEPRECATED'
# define NAN_DEPRECATED __attribute__((deprecated))
                                       ^
../src/unix_dgram.cc:189:37: error: too few arguments to function call, single argument 'context' was not specified
  domain      = info[0]->Int32Value();
                ~~~~~~~~~~~~~~~~~~~ ^
/Users/zaz/Library/Caches/node-gyp/12.8.1/include/node/v8.h:2576:3: note: 'Int32Value' declared here
  V8_WARN_UNUSED_RESULT Maybe<int32_t> Int32Value(Local<Context> context) const;
  ^
/Users/zaz/Library/Caches/node-gyp/12.8.1/include/node/v8config.h:351:31: note: expanded from macro 'V8_WARN_UNUSED_RESULT'
#define V8_WARN_UNUSED_RESULT __attribute__((warn_unused_result))
                              ^
../src/unix_dgram.cc:190:37: error: too few arguments to function call, single argument 'context' was not specified
  type        = info[1]->Int32Value();
                ~~~~~~~~~~~~~~~~~~~ ^
/Users/zaz/Library/Caches/node-gyp/12.8.1/include/node/v8.h:2576:3: note: 'Int32Value' declared here
  V8_WARN_UNUSED_RESULT Maybe<int32_t> Int32Value(Local<Context> context) const;
  ^
/Users/zaz/Library/Caches/node-gyp/12.8.1/include/node/v8config.h:351:31: note: expanded from macro 'V8_WARN_UNUSED_RESULT'
#define V8_WARN_UNUSED_RESULT __attribute__((warn_unused_result))
                              ^
../src/unix_dgram.cc:191:37: error: too few arguments to function call, single argument 'context' was not specified
  protocol    = info[2]->Int32Value();
                ~~~~~~~~~~~~~~~~~~~ ^
/Users/zaz/Library/Caches/node-gyp/12.8.1/include/node/v8.h:2576:3: note: 'Int32Value' declared here
  V8_WARN_UNUSED_RESULT Maybe<int32_t> Int32Value(Local<Context> context) const;
  ^
/Users/zaz/Library/Caches/node-gyp/12.8.1/include/node/v8config.h:351:31: note: expanded from macro 'V8_WARN_UNUSED_RESULT'
#define V8_WARN_UNUSED_RESULT __attribute__((warn_unused_result))
                              ^
../src/unix_dgram.cc:230:28: error: too few arguments to function call, single argument 'context' was not specified
  fd = info[0]->Int32Value();
       ~~~~~~~~~~~~~~~~~~~ ^
/Users/zaz/Library/Caches/node-gyp/12.8.1/include/node/v8.h:2576:3: note: 'Int32Value' declared here
  V8_WARN_UNUSED_RESULT Maybe<int32_t> Int32Value(Local<Context> context) const;
  ^
/Users/zaz/Library/Caches/node-gyp/12.8.1/include/node/v8config.h:351:31: note: expanded from macro 'V8_WARN_UNUSED_RESULT'
#define V8_WARN_UNUSED_RESULT __attribute__((warn_unused_result))
                              ^
../src/unix_dgram.cc:231:21: error: no matching constructor for initialization of 'String::Utf8Value'
  String::Utf8Value path(info[1]);
                    ^    ~~~~~~~
/Users/zaz/Library/Caches/node-gyp/12.8.1/include/node/v8.h:3009:5: note: candidate constructor not viable: no known conversion from 'v8::Local<v8::Value>' to 'const v8::String::Utf8Value' for 1st argument
    Utf8Value(const Utf8Value&) = delete;
    ^
/Users/zaz/Library/Caches/node-gyp/12.8.1/include/node/v8.h:3002:5: note: candidate constructor not viable: requires 2 arguments, but 1 was provided
    Utf8Value(Isolate* isolate, Local<v8::Value> obj);
    ^
../src/unix_dgram.cc:258:28: error: too few arguments to function call, single argument 'context' was not specified
  fd = info[0]->Int32Value();
       ~~~~~~~~~~~~~~~~~~~ ^
/Users/zaz/Library/Caches/node-gyp/12.8.1/include/node/v8.h:2576:3: note: 'Int32Value' declared here
  V8_WARN_UNUSED_RESULT Maybe<int32_t> Int32Value(Local<Context> context) const;
  ^
/Users/zaz/Library/Caches/node-gyp/12.8.1/include/node/v8config.h:351:31: note: expanded from macro 'V8_WARN_UNUSED_RESULT'
#define V8_WARN_UNUSED_RESULT __attribute__((warn_unused_result))
                              ^
../src/unix_dgram.cc:259:18: error: no matching member function for call to 'ToObject'
  buf = info[1]->ToObject();
        ~~~~~~~~~^~~~~~~~
/Users/zaz/Library/Caches/node-gyp/12.8.1/include/node/v8.h:2539:44: note: candidate function not viable: requires single argument 'context', but no arguments were provided
  V8_WARN_UNUSED_RESULT MaybeLocal<Object> ToObject(
                                           ^
/Users/zaz/Library/Caches/node-gyp/12.8.1/include/node/v8.h:2553:31: note: candidate function not viable: requires single argument 'isolate', but no arguments were provided
                Local<Object> ToObject(Isolate* isolate) const);
                              ^
../src/unix_dgram.cc:260:33: error: too few arguments to function call, single argument 'context' was not specified
  offset = info[2]->Uint32Value();
           ~~~~~~~~~~~~~~~~~~~~ ^
/Users/zaz/Library/Caches/node-gyp/12.8.1/include/node/v8.h:2574:3: note: 'Uint32Value' declared here
  V8_WARN_UNUSED_RESULT Maybe<uint32_t> Uint32Value(
  ^
/Users/zaz/Library/Caches/node-gyp/12.8.1/include/node/v8config.h:351:31: note: expanded from macro 'V8_WARN_UNUSED_RESULT'
#define V8_WARN_UNUSED_RESULT __attribute__((warn_unused_result))
                              ^
../src/unix_dgram.cc:261:33: error: too few arguments to function call, single argument 'context' was not specified
  length = info[3]->Uint32Value();
           ~~~~~~~~~~~~~~~~~~~~ ^
/Users/zaz/Library/Caches/node-gyp/12.8.1/include/node/v8.h:2574:3: note: 'Uint32Value' declared here
  V8_WARN_UNUSED_RESULT Maybe<uint32_t> Uint32Value(
  ^
/Users/zaz/Library/Caches/node-gyp/12.8.1/include/node/v8config.h:351:31: note: expanded from macro 'V8_WARN_UNUSED_RESULT'
#define V8_WARN_UNUSED_RESULT __attribute__((warn_unused_result))
                              ^
../src/unix_dgram.cc:262:21: error: no matching constructor for initialization of 'String::Utf8Value'
  String::Utf8Value path(info[4]);
                    ^    ~~~~~~~
/Users/zaz/Library/Caches/node-gyp/12.8.1/include/node/v8.h:3009:5: note: candidate constructor not viable: no known conversion from 'v8::Local<v8::Value>' to 'const v8::String::Utf8Value' for 1st argument
    Utf8Value(const Utf8Value&) = delete;
    ^
/Users/zaz/Library/Caches/node-gyp/12.8.1/include/node/v8.h:3002:5: note: candidate constructor not viable: requires 2 arguments, but 1 was provided
    Utf8Value(Isolate* isolate, Local<v8::Value> obj);
    ^
../src/unix_dgram.cc:302:28: error: too few arguments to function call, single argument 'context' was not specified
  fd = info[0]->Int32Value();
       ~~~~~~~~~~~~~~~~~~~ ^
/Users/zaz/Library/Caches/node-gyp/12.8.1/include/node/v8.h:2576:3: note: 'Int32Value' declared here
  V8_WARN_UNUSED_RESULT Maybe<int32_t> Int32Value(Local<Context> context) const;
  ^
/Users/zaz/Library/Caches/node-gyp/12.8.1/include/node/v8config.h:351:31: note: expanded from macro 'V8_WARN_UNUSED_RESULT'
#define V8_WARN_UNUSED_RESULT __attribute__((warn_unused_result))
                              ^
../src/unix_dgram.cc:303:18: error: no matching member function for call to 'ToObject'
  buf = info[1]->ToObject();
        ~~~~~~~~~^~~~~~~~
/Users/zaz/Library/Caches/node-gyp/12.8.1/include/node/v8.h:2539:44: note: candidate function not viable: requires single argument 'context', but no arguments were provided
  V8_WARN_UNUSED_RESULT MaybeLocal<Object> ToObject(
                                           ^
/Users/zaz/Library/Caches/node-gyp/12.8.1/include/node/v8.h:2553:31: note: candidate function not viable: requires single argument 'isolate', but no arguments were provided
                Local<Object> ToObject(Isolate* isolate) const);
                              ^
../src/unix_dgram.cc:340:28: error: too few arguments to function call, single argument 'context' was not specified
  fd = info[0]->Int32Value();
       ~~~~~~~~~~~~~~~~~~~ ^
/Users/zaz/Library/Caches/node-gyp/12.8.1/include/node/v8.h:2576:3: note: 'Int32Value' declared here
  V8_WARN_UNUSED_RESULT Maybe<int32_t> Int32Value(Local<Context> context) const;
  ^
/Users/zaz/Library/Caches/node-gyp/12.8.1/include/node/v8config.h:351:31: note: expanded from macro 'V8_WARN_UNUSED_RESULT'
#define V8_WARN_UNUSED_RESULT __attribute__((warn_unused_result))
                              ^
../src/unix_dgram.cc:341:21: error: no matching constructor for initialization of 'String::Utf8Value'
  String::Utf8Value path(info[1]);
                    ^    ~~~~~~~
/Users/zaz/Library/Caches/node-gyp/12.8.1/include/node/v8.h:3009:5: note: candidate constructor not viable: no known conversion from 'v8::Local<v8::Value>' to 'const v8::String::Utf8Value' for 1st argument
    Utf8Value(const Utf8Value&) = delete;
    ^
/Users/zaz/Library/Caches/node-gyp/12.8.1/include/node/v8.h:3002:5: note: candidate constructor not viable: requires 2 arguments, but 1 was provided
    Utf8Value(Isolate* isolate, Local<v8::Value> obj);
    ^
../src/unix_dgram.cc:361:28: error: too few arguments to function call, single argument 'context' was not specified
  fd = info[0]->Int32Value();
       ~~~~~~~~~~~~~~~~~~~ ^
/Users/zaz/Library/Caches/node-gyp/12.8.1/include/node/v8.h:2576:3: note: 'Int32Value' declared here
  V8_WARN_UNUSED_RESULT Maybe<int32_t> Int32Value(Local<Context> context) const;
  ^
/Users/zaz/Library/Caches/node-gyp/12.8.1/include/node/v8config.h:351:31: note: expanded from macro 'V8_WARN_UNUSED_RESULT'
#define V8_WARN_UNUSED_RESULT __attribute__((warn_unused_result))
                              ^
../src/unix_dgram.cc:395:11: warning: 'Set' is deprecated: Use maybe version [-Wdeprecated-declarations]
  target->Set(Nan::New(\"AF_UNIX\").ToLocalChecked(), Nan::New(AF_UNIX));
          ^
/Users/zaz/Library/Caches/node-gyp/12.8.1/include/node/v8.h:3365:3: note: 'Set' has been explicitly marked deprecated here
  V8_DEPRECATE_SOON(\"Use maybe version\",
  ^
/Users/zaz/Library/Caches/node-gyp/12.8.1/include/node/v8config.h:326:29: note: expanded from macro 'V8_DEPRECATE_SOON'
  declarator __attribute__((deprecated(message)))
                            ^
../src/unix_dgram.cc:396:11: warning: 'Set' is deprecated: Use maybe version [-Wdeprecated-declarations]
  target->Set(Nan::New(\"SOCK_DGRAM\").ToLocalChecked(), Nan::New(SOCK_DGRAM));
          ^
/Users/zaz/Library/Caches/node-gyp/12.8.1/include/node/v8.h:3365:3: note: 'Set' has been explicitly marked deprecated here
  V8_DEPRECATE_SOON(\"Use maybe version\",
  ^
/Users/zaz/Library/Caches/node-gyp/12.8.1/include/node/v8config.h:326:29: note: expanded from macro 'V8_DEPRECATE_SOON'
  declarator __attribute__((deprecated(message)))
                            ^
../src/unix_dgram.cc:399:63: error: too few arguments to function call, single argument 'context' was not specified
              Nan::New<FunctionTemplate>(Socket)->GetFunction());
              ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ ^
/Users/zaz/Library/Caches/node-gyp/12.8.1/include/node/v8.h:5955:3: note: 'GetFunction' declared here
  V8_WARN_UNUSED_RESULT MaybeLocal<Function> GetFunction(
  ^
/Users/zaz/Library/Caches/node-gyp/12.8.1/include/node/v8config.h:351:31: note: expanded from macro 'V8_WARN_UNUSED_RESULT'
#define V8_WARN_UNUSED_RESULT __attribute__((warn_unused_result))
                              ^
../src/unix_dgram.cc:402:61: error: too few arguments to function call, single argument 'context' was not specified
              Nan::New<FunctionTemplate>(Bind)->GetFunction());
              ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ ^
/Users/zaz/Library/Caches/node-gyp/12.8.1/include/node/v8.h:5955:3: note: 'GetFunction' declared here
  V8_WARN_UNUSED_RESULT MaybeLocal<Function> GetFunction(
  ^
/Users/zaz/Library/Caches/node-gyp/12.8.1/include/node/v8config.h:351:31: note: expanded from macro 'V8_WARN_UNUSED_RESULT'
#define V8_WARN_UNUSED_RESULT __attribute__((warn_unused_result))
                              ^
../src/unix_dgram.cc:405:63: error: too few arguments to function call, single argument 'context' was not specified
              Nan::New<FunctionTemplate>(SendTo)->GetFunction());
              ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ ^
/Users/zaz/Library/Caches/node-gyp/12.8.1/include/node/v8.h:5955:3: note: 'GetFunction' declared here
  V8_WARN_UNUSED_RESULT MaybeLocal<Function> GetFunction(
  ^
/Users/zaz/Library/Caches/node-gyp/12.8.1/include/node/v8config.h:351:31: note: expanded from macro 'V8_WARN_UNUSED_RESULT'
#define V8_WARN_UNUSED_RESULT __attribute__((warn_unused_result))
                              ^
../src/unix_dgram.cc:408:61: error: too few arguments to function call, single argument 'context' was not specified
              Nan::New<FunctionTemplate>(Send)->GetFunction());
              ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ ^
/Users/zaz/Library/Caches/node-gyp/12.8.1/include/node/v8.h:5955:3: note: 'GetFunction' declared here
  V8_WARN_UNUSED_RESULT MaybeLocal<Function> GetFunction(
  ^
/Users/zaz/Library/Caches/node-gyp/12.8.1/include/node/v8config.h:351:31: note: expanded from macro 'V8_WARN_UNUSED_RESULT'
#define V8_WARN_UNUSED_RESULT __attribute__((warn_unused_result))
                              ^
fatal error: too many errors emitted, stopping now [-ferror-limit=]
4 warnings and 20 errors generated.
make: *** [Release/obj.target/unix_dgram/src/unix_dgram.o] Error 1
gyp ERR! build error 
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/Users/zaz/.nvm/versions/node/v12.8.1/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:196:23)
gyp ERR! stack     at ChildProcess.emit (events.js:203:13)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:272:12)
gyp ERR! System Darwin 18.7.0
gyp ERR! command \"/Users/zaz/.nvm/versions/node/v12.8.1/bin/node\" \"/Users/zaz/.nvm/versions/node/v12.8.1/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js\" \"rebuild\"
gyp ERR! cwd /Users/zaz/D**s/A**s/api/node_modules/unix-dgram
gyp ERR! node -v v12.8.1
gyp ERR! node-gyp -v v5.0.3
[6/6] ♻️  Cleaning modules...
warning api@0.0.0: No license field
success Saved 1 new dependency.
info Direct dependencies
└─ winston-syslog@2.3.0
info All dependencies
└─ winston-syslog@2.3.0
✨  Done in 16.65s.
```

**Comment by tsondergaard** [comment]
winston-syslog pulls in unix-dgram@2.0.2. The build issue with unix-dgram is fixed in 2.0.3, so it should be easy to fix

**Comment by tsondergaard** [comment]
I can see in the master branch that upgrading to unix-dgram 2.0.3 was already attempted in commit 41cab43379665b. It caused problems and had to be reverted (commit 378d0130f2d690).

The hang issue is reported here: https://github.com/bnoordhuis/node-unix-dgram/issues/52

**Comment by brenc** [comment]
I encountered this issue too and it almost blocked upgrading to Node 12.x completely. Looks like a workaround is adding `unix-dgram@2.0.3` to your project directly.

**Comment by DABH** [comment]
Finally fixed this it seems, as of `winston-syslog@v2.4.0` (just published).  Thanks all.
